Scripts/script-TaniumMultiHash.yml

```diff
@@ -0,0 +1,184 @@
+      RES_UNAV2= "[results currently unavailable]"
+      NOT_FOUND = "No Matches Found"
+      df_merge = pd.concat([df_dir,df_file,df_md5,df_sha256],axis=1)
+      df_filtered = df_merge[(df_merge['file'] != RES_UNAV2 ) &
+                             (df_merge['file'] != RES_UNAV ) &
+                             (df_merge['file'] != NOT_FOUND) &
+                             (df_merge['file'] != "")]
+      return df_filtered.to_dict('records')
+
+  md5_array = demisto.args()["hash_array"]
+  complete_pct = demisto.args()["complete_pct"]
+  page_length = int(demisto.args()["page_length"])
+
+  if isinstance(md5_array,list):
+      pass
+  else:
+      md5_array=[md5_array]
+
+  paged_list = [md5_array[i:i+page_length] for i in range(0,len(md5_array),page_length)]
+  hash_type = demisto.args()["hash_type"]
+  if hash_type=="MD5":
+      param_hash ="fileMD5Hash"
+  elif hash_type=="SHA256":
+      param_hash = "fileSHA256Hash"
+  for page in paged_list:
+      cpu_col = "Computer Name;Computer ID"
+      file_cols = "Index Query File Details{{"+param_hash+"={0},limit=1}}"
+      #file_cols = "Index Query File Details{{"+param_hash+"={0}}}"
+      question_filters = "Index Query File Exists{{"+param_hash+"={0}}},that equals:Yes"
+      all_file_cols = [file_cols.format(md5) for md5 in page]
+      all_cols = cpu_col+";"+";".join(all_file_cols)
+      all_filters=[question_filters.format(md5) for md5 in page]
+      all_filters_string=";".join(all_filters)
+
+      #demisto.info("all:"+all_cols)
+      tn_manual={}
+      tn_manual["sensors"]=all_cols
+      tn_manual["question_filters"]=all_filters_string
+      tn_manual["complete_pct"]=complete_pct
+      tn_manual["question_options"]="or"
+      hashes = {"hashes":page}
+      demisto.results([{"Type": entryTypes["note"],
+                  "ContentsFormat": formats["json"],
+                  "Contents": hashes,
+                  "HumanReadable":tableToMarkdown("Searching hashes:",hashes)
+
+              }])
+      demisto.info("demisto page hashes: " + str(page))
+      #demisto.log("tanium-question: "+str(tn_manual))
+      demisto.info("tanium-question: "+str(tn_manual))
+      res = demisto.executeCommand("tn-ask-manual-question",tn_manual)
+      #demisto.info("res:" + str(res))
+      #for debugging xml response
+      #demisto.log(res)
+      #sys.exit()
+      #error check
+      if res is None:
+          demisto.info("error res is None")
+          demisto.results([{"Type": entryTypes["error"], "ContentsFormat": formats["text"],
+                              "Contents": "Result from tanium question is None; Error"}])
+          continue
+      result = res[0]
+      if isError(result):
+          demisto.info("is error")
+          demisto.results(res)
+          continue
+      demisto.info("not error")
+
+      contents = result["Contents"]
+      human_readable = result["HumanReadable"]
+      #contents exist check
+      if not contents:
```

Comment:
Not sure why are we doing all those checks if you send it to demisto.results anyway? Perhaps you should raise an exception if you have an error?

Reply:
Raising an exception does not return an error to our war room that i can see. This may have been fixed in a newer version of Demisto though.